devito/stencilkernel.py

```diff
@@ -97,7 +98,8 @@ def __init__(self, stencils, **kwargs):
 
         # Introduce C-level profiling infrastructure
         self.sections = OrderedDict()
-        nodes = self._profile_sections(nodes)
+        if self.profiler:
```

Comment:
To help you with this exercise, I pushed a commit that does most of the stuff in the branch ``stencil-kernel-basic``, based on top of this one. I didn't even try to run it, it's just there to hopefully help you understanding what's the structure that I'd like to obtain eventually